Fatras/include/ActsFatras/Digitization/Channelizer.hpp

```diff
@@ -8,97 +8,66 @@
-                const Acts::Vector2& start)
-        : delta(delta_),
-          intersect(std::move(intersect_)),
-          path((intersect - start).norm()) {}
+    auto maskedSegmentRes = m_surfaceMask.apply(surface, driftedSegment);
 
-    /// Smaller operator for sorting the ChannelStep objects.
-    ///
-    /// @param cstep The other ChannelStep to be compared
-    ///
-    /// The ChannelStep objects can be compared with its path distance
-    /// from the start (surface segment origin)
-    bool operator<(const ChannelStep& cstep) const { return path < cstep.path; }
-  };
+    if (!maskedSegmentRes.ok()) {
+      return maskedSegmentRes.error();
+    }
 
-  /// Nested struct for representing channel steps.
-  struct ChannelSegment {
-    /// The bin of this segment
-    Bin2D bin = {0, 0};
-    /// The segment start, end points
-    Segment2D path2D;
-    /// The (clipped) value (uncorrected: path length)
-    double activation = 0.;
+    // Now Channelize
+    auto segments =
+        m_segmentizer.segments(gctx, surface, segmentation, *maskedSegmentRes);
 
-    /// Constructor with arguments
-    ///
-    /// @param bin_ The bin corresponding to this step
-    /// @param path2D_ The start/end 2D position of the segement
-    /// @param activation_ The segment activation (clean: length) for this bin
-    ChannelSegment(Bin2D bin_, Segment2D path2D_, double activation_)
-        : bin(bin_), path2D(std::move(path2D_)), activation(activation_) {}
-  };
+    // Go from 2D-path to 3D-path by applying thickness
+    const auto path2D = std::accumulate(
+        segments.begin(), segments.end(), 0.0,
+        [](double sum, const auto& seg) { return sum + seg.activation; });
 
-  /// Divide the surface segment into channel segments.
-  ///
-  /// @note Channelizing is done in cartesian coordinates (start/end)
-  /// @note The start and end cartesian vector is supposed to be inside
-  /// the surface bounds (pre-run through the SurfaceMasker)
-  /// @note The segmentation has to be 2-dimensional, even if the
-  /// actual readout is 1-dimensional, in latter case one bin in the
-  /// second coordinate direction is required.
-  ///
-  /// @param geoCtx The geometry context for the localToGlobal, etc.
-  /// @param surface The surface for the channelizing
-  /// @param segmentation The segmentation for the channelizing
-  /// @param segment The surface segment (cartesian coordinates)
-  ///
-  /// @return a vector of ChannelSegment objects
-  std::vector<ChannelSegment> segments(const Acts::GeometryContext& geoCtx,
-                                       const Acts::Surface& surface,
-                                       const Acts::BinUtility& segmentation,
-                                       const Segment2D& segment) const;
+    for (auto& seg : segments) {
+      auto r = path2D != 0.0 ? (seg.activation / path2D) : 1.0;
+      auto segThickness = r * thickness;
+
+      seg.activation = std::hypot(segThickness, seg.activation);
```

Comment:
Is there a unit test for this?
I.e. that the 3D path length from the segments add up to the full 3D path after segmentation?